docs/design/services/s3/transfermanager/README.md

```diff
@@ -0,0 +1,67 @@
+# Project Tenets (unless you know better ones)
+
+1. Meeting customers in their problem space allows them to deliver value
+   quickly.
+2. Meeting customer expectations drives usability.
+3. Discoverability drives usage.
+
+# Introduction
+
+This project provides a much improved experience for S3 customers needing to
+easily perform uploads and downloads of objects to and from S3 by providing
+the S3 `TransferManager`, a high level library built on the S3 client.
+
+# Project Goals
+
+1. For the use cases it addresses, i.e. the transfer of objects to and from S3,
+   TransferManager is the preferred solution. It is easier and more intuitive
+   than using the S3 client. In the majority of situations, it is more
+   performant.
+1. TransferManager provides a truly asynchronous, non-blocking API that conforms
+   to the norms present in the rest of the SDK.
+1. TransferManager makes efficient use of system resources.
+1. TransferManager performs well in cold JVMs.
```

Comment:
I'm not sure if this actually needs to be a project goal. Once you get to the point where using Transfer Manager is more performant, a slightly increased cold start won't likely be a big deal.

Reply:
And to further clarify, I don't think this is actually going to be an issue in Transfer Manager but if for some weird reason we needed to use something like ObjectMapper I don't think that'd be a huge deal for this case.

Reply:
Yeah that sounds fine to me. As you said, I don't foresee the implementation doing anything that would increase cold start over the baseline.